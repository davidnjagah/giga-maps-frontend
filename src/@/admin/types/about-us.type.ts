export interface AboutUsImage {
  id: number;
  image: string;
  name: string;
}

export interface AboutusFormType {
  sectionName: string
  type: string
  fields: AboutFormFieldType[]
  content?: AboutUsContentType
}

export interface AboutFormFieldType {
  name: string
  path: string
  typeField: AboutUsFieldType
}

export interface AboutUsFieldType {
  type: string
  props?: Record<string, string>
  value?: string
  listType?: string
}

export interface AboutUsContentType {
  name: string
  path: string
  allowNew: boolean
  items: Item[]
  maxAllow?: number
}

export interface Item {
  type: string
  fields: AboutFormFieldType[]
}


