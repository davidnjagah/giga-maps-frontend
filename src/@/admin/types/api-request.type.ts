export interface ApiKeysAdminRequestType {
  id: number
  api_key: string
  valid_from: string
  valid_to: string
  user: User
  api: Api
  filters: Filters
  status: string
  extension_status: string
  extension_valid_to: string;
  status_updated_by: unknown
  created: string
  last_modified_at: string
  is_active: boolean
  active_countries_list: {
    id: number
    name: string;
  }[]
}

interface User {
  id: number
  first_name: string
  last_name: string
  email: string
}

interface Api {
  id: number
  name: string
  category: string
}

interface Filters { }