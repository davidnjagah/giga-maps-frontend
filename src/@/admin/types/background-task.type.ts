export interface BackgroundTask {
  task_id: string;
  status: string;
  log: string;
  created_at: string;
  completed_at: string;
}