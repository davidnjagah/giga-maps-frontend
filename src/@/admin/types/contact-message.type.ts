export interface ContactMessage {
  id: number;
  created: string;
  modified: string;
  full_name: string;
  organisation: string;
  purpose: string;
  message: string
}