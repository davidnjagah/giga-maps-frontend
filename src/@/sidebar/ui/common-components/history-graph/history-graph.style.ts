import { styled } from "styled-components";

import { Scroll } from "~/@/scroll";

export const HistoryGraphWrapper = styled.div<{ $isWeek: boolean; }>`
  overflow: hidden;
  .history-modal__period-unit-picker {
    margin-bottom: 1rem;
    margin-top: 0.75rem;

    .history-modal__period-unit.history-modal__period-unit--active {
      display: flex;
      height: 2rem;
      padding: 0.6875rem 1rem;
      justify-content: center;
      align-items: center;
      gap: 0.5rem;
      border-bottom: 2px solid  ${props => props.theme.graphWeekMonthBorder};
      // background: var(--transparent-transparent, ${props => props.theme.text});
      // background-blend-mode: multiply;
      text-transform:none;
    }

    .history-modal__period-unit {
      display: flex;
      width: 4.75rem;
      height: 2rem;
      padding: 0.4375rem 1rem;
      justify-content: center;
      align-items: center;
      color: ${props => props.theme.graphWeekMonthBorder};
      border-bottom: 2px solid  ${props => props.theme.graphWeekMonthBorder};
      // background: var(--transparent-transparent, ${props => props.theme.text});
      // background-blend-mode: multiply;
      text-transform:none;
      font-family: inherit;
    }
  }

  .history-graph-lineChart {
    /* margin: 0 1rem; */
    padding: 1rem 1rem 0 0;
    position: relative;

    .chart-holder.cds--chart-holder {
      /* width: 16.5rem !important; */
      .cds--cc--toolbar, .cds--cc--layout-row {
        /* display:none; */
      }
    }

   
  }

  .history-graph-lineChart {
.bx--chart-holder .bx--cc--toolbar{
  display: none;
}
.bx--chart-holder{
  margin: 0 0.5rem;
  height: inherit;
  width: 22rem;
}
}


.history-modal__period-unit-picker{
  display: flex;
  flex-direction: row;
  align-items: center;
  width: inherit;
  margin: 0;
  > button{
    padding: 0.6rem;
    color: #b6bbc6;
    font-weight: bold;
    font-size: 0.6rem;
    font-family: "Cabin", sans-serif;
    text-transform: uppercase;
    background-color: transparent;
    border: none;
    outline: none;
    cursor: pointer;
    width: 50%;
  }
}


.bx--cc--chart-wrapper text{
  font-size: 0.6rem;
}


.history-modal__period-unit:hover{
background-color: unset;
}

.history-modal__period-unit-picker > .history-modal__period-unit.history-modal__period-unit--active{
  color: ${props => props.theme.text};
  display: flex;
  height: 2rem;
  padding: 0.6875rem 1rem;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
  border-bottom: 2px solid  ${props => props.theme.text};
  // background: var(--transparent-transparent, ${props => props.theme.text});
  // background-blend-mode: multiply;
  font-family: inherit;
}

.history-modal__period-unit-picker{
  /* margin-left: 1rem; */
  width: 50%;
}

.chart-grid-backdrop {
  display: none;
}

.cds--cc--threshold line.threshold-line {
  stroke-width: 2px;
  stroke-dasharray: 0;
}

`

export const BarChartScrollable = styled(Scroll)`
  overflow-y: hidden;
  padding-right: 1.7rem;
`

export const PanExpander = styled.div`
  position: absolute;
  z-index: 2;
  right: 16px;
  top:-16px;
  svg {
    fill: ${props => props.theme.text} !important;
  }
`