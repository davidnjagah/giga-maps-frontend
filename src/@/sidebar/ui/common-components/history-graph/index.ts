export { default } from './history-graph.view'
export * from './history-graph-accordian.view'