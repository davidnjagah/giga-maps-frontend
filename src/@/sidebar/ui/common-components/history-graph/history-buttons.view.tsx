import clsx from "clsx";
import { changeHistoryIntervalUnit } from "~/@/sidebar/history-graph.model";
import { IntervalUnit } from "~/lib/date-fns-kit/types";


export default function HistoryButtons({ isWeek }: { readonly isWeek: boolean }) {
  return <div className="history-modal__period-unit-picker">
    <button
      type="button"
      onClick={() => {
        changeHistoryIntervalUnit(IntervalUnit.week);
      }}
      className={clsx('history-modal__period-unit', {
        'history-modal__period-unit--active': isWeek,
      })}
    >
      Weekly
    </button>
    <button
      type="button"
      onClick={() => changeHistoryIntervalUnit(IntervalUnit.month)}
      className={clsx('history-modal__period-unit', {
        'history-modal__period-unit--active': !isWeek,
      })}
    >
      Monthly
    </button>
  </div>
}