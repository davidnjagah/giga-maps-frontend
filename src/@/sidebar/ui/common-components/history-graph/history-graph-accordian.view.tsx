import { ChartColumn, } from "@carbon/icons-react";
import { AccordionItem } from "@carbon/react";
import { useState } from "react";
import styled from "styled-components";

import { SchoolStatsType } from "~/api/types";

import { AccordionItemTitle } from "../../global-and-country-view-components/common/accordion-item-title.view";
import { AccordionDistribution } from "../../sidebar.style";
import HistoryGraph from "./history-graph.view";

const ProgressGraphWrapper = styled.div`
    margin: 0rem 1rem 0rem 1rem;

`

export function HistoryGraphAccordian({ schoolData, isLoading }: { schoolData?: SchoolStatsType; isLoading?: boolean }) {
  const [show, setShow] = useState(true);
  return (
    // <ProgressGraphWrapper>
    //   <AccordionDistribution>
    //     <AccordionItem
    //       title={
    //         <AccordionItemTitle
    //           label=
    //           {
    //             <>
    //               {/* <ChartColumn className="graph" /> */}
    //               Progress graph
    //             </>
    //           }
    //           tooltipLabel="Progress graph" />}
    //       open={show}
    //       onHeadingClick={() => setShow((prev) => !prev)}>
    //       <HistoryGraph isChartOpen={true} schoolData={schoolData} isLoading={isLoading} />
    //     </AccordionItem>
    //   </AccordionDistribution>
    // </ProgressGraphWrapper>
    <ProgressGraphWrapper><HistoryGraph isChartOpen={true} schoolData={schoolData} isLoading={isLoading} /></ProgressGraphWrapper>
  )
}