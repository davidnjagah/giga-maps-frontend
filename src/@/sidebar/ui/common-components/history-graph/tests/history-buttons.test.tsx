import { render, screen } from '@testing-library/react';
import HistoryButtons from '../history-buttons.view';
import userEvent from '@testing-library/user-event';
import { changeHistoryIntervalUnit } from '~/@/sidebar/history-graph.model';
import { IntervalUnit } from '~/lib/date-fns-kit/types';

describe('Layer schools connectivity status', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders "Weekly" and "Monthly" buttons correctly', () => {
    render(<HistoryButtons isWeek={true} />);
    const weeklyButton = screen.getByText('Weekly');
    const monthlyButton = screen.getByText('Monthly');
    expect(weeklyButton).toMatchSnapshot();
    expect(monthlyButton).toMatchSnapshot();
  });

  it('calls changeHistoryIntervalUnit with IntervalUnit.week when "Weekly" button is clicked', () => {
    render(<HistoryButtons isWeek={true} />);

    const weeklyButton = screen.getByText('Weekly');
    expect(weeklyButton).toMatchSnapshot();
  });

  it('calls changeHistoryIntervalUnit with IntervalUnit.month when "Monthly" button is clicked', () => {
    render(<HistoryButtons isWeek={false} />);
    const monthlyButton = screen.getByText('Monthly');
    expect(monthlyButton).toMatchSnapshot();
  });
});
