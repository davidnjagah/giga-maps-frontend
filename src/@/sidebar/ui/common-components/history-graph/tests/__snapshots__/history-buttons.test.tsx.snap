// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer schools connectivity status calls changeHistoryIntervalUnit with IntervalUnit.month when "Monthly" button is clicked 1`] = `
<button
  class="history-modal__period-unit history-modal__period-unit--active"
  type="button"
>
  Monthly
</button>
`;

exports[`Layer schools connectivity status calls changeHistoryIntervalUnit with IntervalUnit.week when "Weekly" button is clicked 1`] = `
<button
  class="history-modal__period-unit history-modal__period-unit--active"
  type="button"
>
  Weekly
</button>
`;

exports[`Layer schools connectivity status renders "Weekly" and "Monthly" buttons correctly 1`] = `
<button
  class="history-modal__period-unit history-modal__period-unit--active"
  type="button"
>
  Weekly
</button>
`;

exports[`Layer schools connectivity status renders "Weekly" and "Monthly" buttons correctly 2`] = `
<button
  class="history-modal__period-unit"
  type="button"
>
  Monthly
</button>
`;
